pkg/queue/manager.go

```diff
@@ -547,3 +570,36 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) Start(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx).WithName("clusterQueueSnapshot")
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	ticker := time.NewTicker(
+		time.Duration(m.cfg.QueueVisibility.UpdateIntervalSeconds) * time.Second,
+	)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.V(2).Info("Context cancelled; stop doing snapshot of cluster queue")
+			return nil
+		case t := <-ticker.C:
+			m.workloadsStatus = &kueue.ClusterQueuePendingWorkloadsStatus{
+				Head:           make([]kueue.ClusterQueuePendingWorkload, 0),
+				LastChangeTime: metav1.Time{Time: t},
+			}
+			for _, wl := range m.Snapshot() {
+				m.workloadsStatus.Head = append(m.workloadsStatus.Head, kueue.ClusterQueuePendingWorkload{
+					Name:      wl.Name,
+					Namespace: wl.Namespace,
+				})
+			}
+		}
+	}
+}
+
+func (m *Manager) GetWorkloadsStatus() *kueue.ClusterQueuePendingWorkloadsStatus {
+	return m.workloadsStatus
```

Comment:
the lock mentioned above should be locked for read here.

Reply:
This should be a lock just for `m.workloadStatus`

Reply:
mutex was added